pyvista/utilities/reader.py

```diff
@@ -171,6 +171,51 @@ def get_reader(filename, force_ext=None):
     return Reader(filename)
 
 
+class BaseVTKReader(ABC):
+    """Simulate a VTK reader."""
+
+    def __init__(self):
+        self._data_object = None
+        self._observers: List[Union[int, Callable]] = []
+
+    def SetFileName(self, filename):
+        """Set file name."""
+        self._filename = filename
+
+    def UpdateInformation(self):
+        """Update Information from file."""
+        pass
+
+    def AddObserver(self, event_type, callback):
+        """Add Observer that can be triggered during Update."""
+        self._observers.append([event_type, callback])
+
+    def RemoveObservers(self, *args):
+        """Remove Observer."""
+        self._observers = []
+
+    def GetProgress(self):
+        """GetProgress."""
+        return 0.0
```

Comment:
If subclasses are monitored with `ProgressMonitor`, this hardcoded return value will mess it up

Reply:
I'm not actually sure that this does anything when implemented purely in python.  But, even if I'm wrong, if a subclass can declare progress incrementally, it can override this like GIFReader.

Reply:
If called from Python, this overrides. It just doesn't affect any calls on the C++ side

Reply:
I'm thinking the default could check whether `_data_object is None` and return 1 if so.

Probably mostly bike-shedding comments: Since this is a purely sync single thread method, as implented, the entire `Update` method runs before any in progress state could be observed.  We only call `Update` during the `read` method of `BaseReader`, so the user will only see a change _after_ they've already received the mesh, and it will go from 0.0 to 1.0.